### ISSUE TITLE
Rule Request: Flag `assert(false, …)` and `precondition(false, …)`

### ISSUE THREAD

**Issue by lilyball** [realm/SwiftLint#2288]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

My coworkers keep adding code like

```swift
if (someCondition) {
    assert(false, "Error message goes here")
}
```

While this is necessary in Obj-C, in Swift we should be using `asssertionFailure("Error message")` (and `preconditionFailure("Error message")`) instead. There's literally no reason to be writing `assert(false, …)` beyond lack of knowledge of `assertionFailure(_:)`.

I don't see any configuration necessary here, and this should be opt-out because it's generally useful and the only reason to disable it is because you have hundreds of violations and don't want to fix them.

**Comment by ornithocoder** [comment]
Hi @kballard, do you have plans to work on it? Otherwise I'll take it.

**Comment by kballard** [comment]
@ornithocoder I have no plans on doing this myself, so please go ahead.

**Comment by BasThomas** [comment]
This seems to have been fixed in #3506. :)
